server/routerlicious/packages/lambdas/src/scribe/lambdaFactory.ts

```diff
@@ -162,9 +156,24 @@ export class ScribeLambdaFactory extends EventEmitter implements IPartitionLambd
             }
         } else {
             lastCheckpoint = JSON.parse(document.scribe);
-            const message = `Restoring checkpoint from db. Seq no: ${lastCheckpoint.sequenceNumber}`;
-            context.log?.info(message, { messageMetaData });
-            Lumberjack.info(message, getLumberBaseProperties(documentId, tenantId));
+            const lumberjackProperties = {
+                ...getLumberBaseProperties(documentId, tenantId),
+                lastCheckpointSeqNo: lastCheckpoint.sequenceNumber,
+                logOffset: lastCheckpoint.logOffset,
+                LastCheckpointProtocolSeqNo: lastCheckpoint.protocolState.sequenceNumber
+            }
+
+            Lumberjack.info("Restoring checkpoint from db", lumberjackProperties);
+
+            if (!this.getDeltasViaAlfred)
+            {
+                // Fetch pending ops from scribeDeltas collection
+                const dbMessages =
+                    await this.messageCollection.find({ documentId, tenantId }, { "operation.sequenceNumber": 1 });
+                opMessages = dbMessages.map((dbMessage) => dbMessage.operation);
+            } else if (lastCheckpoint.logOffset !== -1) {
```

Comment:
Is this an optimization that can be moved instead to line 157, substituting the `else`? That is, if `getDeltasViaAlfred` is false, and `lastCheckpoint.logOffset` is -1, then we would still make the call to the ops collection even though nothing will be returned from it